### DISCUSSION TITLE
How can I leave prev task definition as activate status.

### DISCUSSION THREAD

**Discussion by jed-glorang** [aws/copilot-cli#5300] [Q&A] [answered]
Hi. I'm using AWS Copilot Backend Service.
When I deploy my service with deploy command('svc deploy ~`), 
it change status of prev task definition to INACTIVE when ECS Service updated successfully.
But, I want to leave prev task definition to active status.

So, I want to ask two things.

1. Is it possible to leave prev task definition to active status?
    (Related configuration is exists on AWS CodeDeploy ECS Blue/Green deployment set up. But AWS Copliot provides only Rolling Update.)
2. If is possible, how can I do this?
    (aws copilot overrides..?)
    
Thanks for read my discussion :)

**Comment by Lou1415926** [comment, marked as answer]
Yup! You can use [yaml patch](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) to update the `TaskDefinition`'s `UpdateReplacePolicy` to `Retain`:
```yaml
- op: add
  path: /Resources/TaskDefinition/UpdateReplacePolicy
  value: Retain
```

Edit: I will add a bit explanation on why this would work.

The task definition became INACTIVE because when CFN updates the stack, it under the hood `CREATE` a task definition with a new revision, and `DELETE` the old task definition revision. In this case, "delete" means "deregistering" the old task def, resulting in the old task def revision becoming INACTIVE.

By adding  `UpdateReplacePolicy: Retain`, we prevent CFN to do the "delete" step, hence keeping the old task def as ACTIVE.

**Reply by jed-glorang** [reply]
Thanks a lot!!
This is what I wanted to find all day long.😁

If you don't mind, I want to ask one more thing.

From what I know, ECS Fargate pricing is related to CPU, Memory, Storages used by running containers.
Is active status of Task Definitions are affect to ECS Fargate pricing?
(For example, Storage billing for Task Definition storing..?)

**Reply by Lou1415926** [reply]
nope - I believe the task definition is not factored into the pricing!

**Reply by shanmukhabandi** [reply]
> Yup! You can use [yaml patch](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) to update the `TaskDefinition`'s `UpdateReplacePolicy` to `Retain`:
> 
> ```yaml
> - op: add
>   path: /Resources/TaskDefinition/UpdateReplacePolicy
>   value: Retain
> ```
> 
> Edit: I will add a bit explanation on why this would work.
> 
> The task definition became INACTIVE because when CFN updates the stack, it under the hood `CREATE` a task definition with a new revision, and `DELETE` the old task definition revision. In this case, "delete" means "deregistering" the old task def, resulting in the old task def revision becoming INACTIVE.
> 
> By adding `UpdateReplacePolicy: Retain`, we prevent CFN to do the "delete" step, hence keeping the old task def as ACTIVE.

HI @Lou1415926

I updated the same in the manifest file, but when we deployed the service, it was not updated in the stack

**Reply by jed-glorang** [reply]
Did you updated that code on [yaml patch](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) file?
It's different from [manifest](https://aws.github.io/copilot-cli/docs/manifest/overview/) file.
